src/hooks/useCollectionContext.tsx

```diff
@@ -0,0 +1,36 @@
+import React, { ReactChild, useReducer } from 'react'
+import { collectionReducer } from 'utils'
+
+const MyCollectionContext = React.createContext<{
+  myCollection: Collection
+  dispatch: React.Dispatch<Action>
+}>({
+  myCollection: { cards: [] },
+  dispatch: () => undefined,
+})
+
+const CollectionProvider = ({
+  children,
+}: {
+  children: ReactChild[] | ReactChild
+}) => {
+  const [myCollection, dispatch] = useReducer(collectionReducer, {
```

Comment:
```suggestion
  const [collection, dispatch] = useReducer(collectionReducer, {
```

I would just name this collection everywhere. Prepending My- for your My Collection page makes since, but I think prepending My- in general is really confusing, I'd avoid it as much as possible.